controllers/clustertopology_controller_compute_test.go

```diff
@@ -17,15 +17,620 @@ limitations under the License.
+	g.Expect(in.obj.GetLabels()).To(HaveKeyWithValue(clusterv1.ClusterTopologyLabelName, ""))
+	for k, v := range in.labels {
+		g.Expect(in.obj.GetLabels()).To(HaveKeyWithValue(k, v))
+	}
+	g.Expect(in.obj.GetAnnotations()).To(HaveKeyWithValue(clusterv1.TemplateClonedFromGroupKindAnnotation, in.templateRef.GroupVersionKind().GroupKind().String()))
+	g.Expect(in.obj.GetAnnotations()).To(HaveKeyWithValue(clusterv1.TemplateClonedFromNameAnnotation, in.templateRef.Name))
+	for k, v := range in.annotations {
+		g.Expect(in.obj.GetAnnotations()).To(HaveKeyWithValue(k, v))
+	}
+	// Spec
+	expectedSpec, ok, err := unstructured.NestedMap(in.template.UnstructuredContent(), "spec", "template", "spec")
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(ok).To(BeTrue())
+
+	cloneSpec, ok, err := unstructured.NestedMap(in.obj.UnstructuredContent(), "spec")
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(ok).To(BeTrue())
+	for k, v := range expectedSpec {
+		g.Expect(cloneSpec).To(HaveKeyWithValue(k, v))
+	}
+}
+
+func assertTemplateToTemplate(g *WithT, in assertTemplateInput) {
+	// TypeMeta
+	g.Expect(in.obj.GetAPIVersion()).To(Equal(in.template.GetAPIVersion()))
+	g.Expect(in.obj.GetKind()).To(Equal(in.template.GetKind()))
+
+	// ObjectMeta
+	if in.currentRef != nil {
+		g.Expect(in.obj.GetName()).To(Equal(in.currentRef.Name))
+	} else {
+		g.Expect(in.obj.GetName()).To(HavePrefix(in.cluster.Name))
+	}
+	g.Expect(in.obj.GetNamespace()).To(Equal(in.cluster.Namespace))
+	g.Expect(in.obj.GetLabels()).To(HaveKeyWithValue(clusterv1.ClusterLabelName, in.cluster.Name))
+	g.Expect(in.obj.GetLabels()).To(HaveKeyWithValue(clusterv1.ClusterTopologyLabelName, ""))
+	for k, v := range in.labels {
+		g.Expect(in.obj.GetLabels()).To(HaveKeyWithValue(k, v))
+	}
+	g.Expect(in.obj.GetAnnotations()).To(HaveKeyWithValue(clusterv1.TemplateClonedFromGroupKindAnnotation, in.templateRef.GroupVersionKind().GroupKind().String()))
+	g.Expect(in.obj.GetAnnotations()).To(HaveKeyWithValue(clusterv1.TemplateClonedFromNameAnnotation, in.templateRef.Name))
+	for k, v := range in.annotations {
+		g.Expect(in.obj.GetAnnotations()).To(HaveKeyWithValue(k, v))
+	}
+	// Spec
+	expectedSpec, ok, err := unstructured.NestedMap(in.template.UnstructuredContent(), "spec")
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(ok).To(BeTrue())
+
+	cloneSpec, ok, err := unstructured.NestedMap(in.obj.UnstructuredContent(), "spec")
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(ok).To(BeTrue())
+	g.Expect(cloneSpec).To(Equal(expectedSpec))
+}
+
+func assertNestedField(g *WithT, obj *unstructured.Unstructured, value interface{}, fields ...string) {
+	v, ok, err := unstructured.NestedFieldCopy(obj.UnstructuredContent(), fields...)
+
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(ok).To(BeTrue())
+	g.Expect(v).To(Equal(value))
+}
+
+func assertNestedFieldDoesNotExists(g *WithT, obj *unstructured.Unstructured, fields ...string) {
```

Comment:
```suggestion
func assertNestedFieldDoesNotExist(g *WithT, obj *unstructured.Unstructured, fields ...string) {
```

Reply:
Or alternatively if you like `assertNestedFieldUnset`

Reply:
Done